Add a note on margin collapsing

aziham opened the pull request:
<!-- 👀 Thanks for opening a PR! Read comments like this one to get your PR merged faster. -->
#### Summary
<!-- ✍️ In a sentence or two, describe your changes -->
I added a note to clarify the intended behavior of the margin collapsing.

#### Motivation
<!-- ❓ Why are you making this change? Help us understand how your changes help readers. -->
People may get the wrong impression that the margin collapsing is supposed to work on all sides of the element. Adding a note to clarify the intended behavior of the margin collapsing will come in handy.

#### Metadata
<!-- ✅ Check a box if applicable, like this: [x]

This PR…
-->
- [ ] Adds a new document
- [x] Rewrites (or significantly expands) a document
- [ ] Fixes a typo, bug, or other error

<!-- 👷‍♀️ After submitting, review the results of the "Checks" tab! -->

aziham committed:
Add a note about the margin collapsing

People may get the wrong impression that the margin collapsing is supposed to work on all sides of the element. Adding a note to clarify the intended behavior of the margin collapsing will come in handy.

OnkarRuikar commented on files/en-us/learn/css/css_layout/normal_flow/index.md:
```diff
@@ -54,6 +54,8 @@ Inline elements behave differently. They don't appear on new lines; instead, the
 
 If two adjacent elements both have a margin set on them and the two margins touch, the larger of the two remains and the smaller one disappears. Recall that this is known as **margin collapsing**.
 
+> **Note:** The margin collapsing only works when the margins of the two adjacent elements touch vertically and not horizontally.
```
Just a suggestion. We can make this concise. [ref](https://drafts.csswg.org/css2/#collapsing-margins)
```suggestion
> **Note:** Horizontal margins never collapse.
```
***
I think we need to mention this also on the dedicated page for the 'margin collapsing' https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Box_Model/Mastering_margin_collapsing.

Hamza-Aziane replied:
Thank you for the suggestion. I agree with what @wbamberg said 

> I think it is more helpful to state the positive side, that only vertical margins collapse.

As it may make it more clear to the reader. But I think that this change should be a note to give the reader that "pay attention" feeling.

Hamza-Aziane replied:
> I think we need to mention this also on the dedicated page for the 'margin collapsing' https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Box_Model/Mastering_margin_collapsing.

Just created a PR for it: #16782

wbamberg commented on files/en-us/learn/css/css_layout/normal_flow/index.md:
```diff
@@ -54,6 +54,8 @@ Inline elements behave differently. They don't appear on new lines; instead, the
 
 If two adjacent elements both have a margin set on them and the two margins touch, the larger of the two remains and the smaller one disappears. Recall that this is known as **margin collapsing**.
 
+> **Note:** The margin collapsing only works when the margins of the two adjacent elements touch vertically and not horizontally.
```
I'm not sure that more concise is necessarily is better here. I think it is more helpful to state the positive side, that only vertical margins collapse.
 
> I think we need to mention this also on the dedicated page for the 'margin collapsing' https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Box_Model/Mastering_margin_collapsing.

Yes, I agree. That's not a great page either :(.

Also, the previous sentence unhelpfully says "**Recall that...**" without linking in case I don't recall. I'm guessing it's expecting us to "recall" from https://developer.mozilla.org/en-US/docs/Learn/CSS/Building_blocks/The_box_model#margin_collapsing, which *also* doesn't mention that margin collapsing only works for vertical margins. But we should link from this page to https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Box_Model/Mastering_margin_collapsing.

Also, I don't think this should be a note. We massively overuse notes on MDN, for reasons.

So perhaps something like:

```suggestion
If two vertically adjacent elements both have a margin set on them and their margins touch, the larger of the two margins remains and the smaller one disappears. This is known as [**margin collapsing**](/en-US/docs/Web/CSS/CSS_Box_Model/Mastering_margin_collapsing).

Margin collapsing only works for vertical (top and bottom) margins, not for horizontal (left and right) margins.
```

OnkarRuikar replied:
> Margin collapsing only works for vertical (top and bottom) margins, not for horizontal (left and right) margins.

This is better. 
It's good that we are not saying it works only for `margin-top` and `margin-bottom` because the `writing-mode` changes things:
```html
<div style="writing-mode:vertical-lr">
  <span style="margin-right:50px">a</span>
  <span style="margin-left:10px">b</span>
</div>
```
https://jsfiddle.net/6hkgm42v/

Hamza-Aziane replied:
Thanks for the suggestion @wbamberg, your suggestion about stating the positive side sounds about right. And as @OnkarRuikar stated:

> It's good that we are not saying it works only for `margin-top` and `margin-bottom` because the `writing-mode` changes things:

For the note part:
> I think that this change should be a note to give the reader that "pay attention" feeling.

Hamza-Aziane replied:
Based on the suggestions from @OnkarRuikar and @wbamberg, maybe something like:

```suggestion
If two adjacent elements both have a margin set on them and the two margins touch, the larger of the two remains and the smaller one disappears. Recall that this is known as **margin collapsing**.

> **Note:** The margin collapsing only works for vertical margins.
```

wbamberg replied:
> I think that this change should be a note to give the reader that "pay attention" feeling.

Please, no, let's not. There are lots of things people have to pay attention to when reading the docs. Having notes for them all really doesn't help readers. The proliferation of nodes on MDN really is a leftover from the Wiki days, where someone would find something missing from the docs, and would decide to add it, and because it's the thing they just spend X hours trying to debug, would think it must be *the most important thing to know about this feature*, so they would add it in a note (or even worse, a warning). The problem is that different people find different things important, so everyone would make their thing note style, and you get a kind of carnival of notes, which destroys any attempt to make a coherent document.

Our job as writers isn't to collect a collection of "must know" things, it's to describe an entire feature in a coherent manner.

Sorry to rant, but this is a bit of a thing for me.

Hamza-Aziane replied:
> The problem is that different people find different things important, so everyone would make their thing note style

I think you've just changed my mind about notes. I was one of the people that assumed margin collapsing is gonna work on all sides; i.e., my motivation for making it a note.

Taking in consideration what @OnkarRuikar demonstrated in his reply:

> It's good that we are not saying it works only for `margin-top` and `margin-bottom` because the `writing-mode` changes things

Any suggestion on what the final edit should be?

Thanks.

Hamza-Aziane replied:
> This doesn't address the problem in [#16745 (comment)](https://github.com/mdn/content/pull/16745#discussion_r884205967) that the previous para says "recall that" without linking to anything.

I totally forgot about this paragraph. I fixed it with my latest commit.

aziham committed:
Update the note on margin collapsing

Rewrite the note on margin collapsing such that it focuses only on the positive side (i.e. vertical margins)

Co-authored-by: wbamberg <will@bootbonnet.ca>

aziham committed:
Merge branch 'mdn:main' into patch-1

aziham committed:
Rewrite the note on margin collapsing

aziham committed:
Rewrite paras on margin collapsing

wbamberg reviewed (approved):
Looks great, thanks @Hamza-Aziane for your contribution!
